### PR TITLE
Marketing Dashboard and Program Detail Optimization by adding the Content Fragment Path

### PR DESCRIPTION
**Pull request by TyroneAEM**
https://experience.adobe.com/#/@wfadoberm/so:adoberm-Production/workfront/task/66c36080001a1f4993ee64439ad4629a/overview

JIRA: ASSETS-00000

Test URLs:
Marketing Dashboard
https://assets-72031--adobe-gmo--hlxsites.hlx.page/qa/marketing-dashboard

Program Detail
https://assets-72031--adobe-gmo--hlxsites.hlx.page/qa/program-details?programName=Acrobat%27s%20Got%20It&programID=6578983c049d6162a756ac83e90f3ccb&path=/content/dam/gmo-cf/en/digital-media-dme/programs/fy24-agi3-0-de-campaign-666017ba00b03fe37af940a93c5a04e2

### CODE REVIEW

File: scripts/graphql.js [hunk 1]
```diff
@@ -57,6 +59,10 @@ export async function graphqlAllCampaignsFilter(first,cursor,filter) {
       },
     };
     const response = await fetch(`${graphqlEndpoint}`, options);
+    //Performance logging
+    const endTime = performance.now();
+    const executionTime = endTime - startTime;
+    console.log(`getAllCampaigns Execution Time: ${executionTime} ms`);
```
Comment (shiv-gup) [inline scripts/graphql.js:65]: @TyroneAEM update the log level to debug

File: scripts/graphql.js [hunk 2]
```diff
@@ -181,6 +189,11 @@ export async function executeQuery(queryString) {
       if (!response.ok) {
           throw new Error(`HTTP error! Status: ${response.status}`);
       }
+      //Performance logging
+      const endTime = performance.now();
+      const executionTime = endTime - startTime;
+      console.log(`executeQuery for ${queryString} Execution Time: ${executionTime} ms`);
```
Comment (shiv-gup) [inline scripts/graphql.js:195]: @TyroneAEM update log level to debug

Comment (TyroneAEM) [reply scripts/graphql.js]: Updated to use console.debug()

File: blocks/gmo-program-details/gmo-program-details.js [hunk 3]
```diff
@@ -1015,20 +1025,33 @@ async function getTaskStatusMapping(taskStatus) {
...
-            const thumbnailDiv = itemEl.querySelector('.thumbnail');
-            const imgElement = document.createElement('img');
-            imgElement.src = imageObject.imageUrl;
-            imgElement.alt = imageObject.imageAltText;
-            imgElement.loading = 'lazy';
-            thumbnailDiv.appendChild(imgElement);
-        } else {
-            console.error("Image Object does not have a valid imageUrl");
+    // Create a unique key for the cache based on the parameters
+    const cacheKey = `${programName}-${campaignName}-${deliverableType}`;
```
Comment (shiv-gup) [inline blocks/gmo-program-details/gmo-program-details.js:1029]: @TyroneAEM can we only add `campaignName` in the `cacheKey` when **campaignName** is not **null** or **empty**

**Review by shiv-gup — APPROVED** [review]

**Pull request merged**